crates/orderbook/src/orderbook.rs

```diff
@@ -312,9 +319,43 @@ impl Orderbook {
         Ok(())
     }
 
+    async fn get_replaced_order(
+        &self,
+        new_order: &OrderCreation,
+    ) -> Result<Order, ReplaceOrderError> {
+        #[derive(Deserialize)]
+        struct AppData {
+            metadata: Metadata,
+        }
+        #[derive(Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        struct Metadata {
+            replaced_order: OrderUid,
```

Comment:
Actually, I have a concern regarding using a flat value for this, which affects this PR too

https://github.com/cowprotocol/app-data/pull/60#discussion_r1505764953